cmd/verify_package.go

```diff
@@ -162,40 +158,174 @@ func verifyPackage(cmd *cobra.Command, args []string) error {
         return err
+    }
+
+    // Verify certificates.
+    if !disableCertificateVerification {
+        // Download root certificate.
+        certPath := filepath.Join(destDir, "ca.crt")
+        if err := downloadRootCert(certPath); err == nil {
+            cmd.Printf("File downloaded at %s\n", certPath)
+        } else {
+            return err
+        }
+
+        if ok, err := verifyCertificate([]byte(sigDetails.CertInfo.CertChain), certPath, cert); ok {
+            cmd.Printf("Certificates verified successfully!\n")
+        } else {
+            cmd.Printf("Unsuccessful Certificate Verification\n")
+            if err != nil {
+                return err
+            }
+        }
+    }
+
+    // Verify data integrity.
+    bytes, err = os.ReadFile(artifactPath)
+    if err != nil {
+        return fmt.Errorf("failed to read artifact file: %v", err)
+    }
+
+    ok, err := verifyDigest(bytes, sigDetails.Digest[0].Digest)
+    if !ok {
+        if err != nil {
+            return err
+        }
+        return fmt.Errorf("incorrect Digest")
+    }
+
+    // Verify authenticity.
+    sig, err := hex.DecodeString(sigDetails.Signature[0].Signature)
+    if err != nil {
+        return fmt.Errorf("failed to decode the hex: %v", err)
+    }
+    dig, err := hex.DecodeString(sigDetails.Digest[0].Digest)
+    if err != nil {
+        return fmt.Errorf("failed to decode the hex: %v", err)
+    }
+    ok, err = verifySignatures([]byte(sig), []byte(dig), cert)
+    if ok {
+        cmd.Println("Signature Verified successfully!")
+    } else {
+        cmd.Println("Unsuccessful Signature Verification")
+        if err != nil {
+            return err
+        }
+    }
+
+    // Verify build provenance.
+    if verifyBuildProvenance {
+        publicKeyPath := filepath.Join(destDir, "public.pem")
+        buildProvSigPath := filepath.Join(destDir, "signature.sig")
+
+        if err := os.WriteFile(buildProvSigPath, buildProvSig, 0644); err != nil {
+            return err
+        }
+        if err := os.WriteFile(publicKeyPath, []byte(provenancePublicKey), 0644); err != nil {
+            return err
+        }
+
+        _, stderror, _, err := verifyBuildProv(publicKeyPath, buildProvSigPath, artifactPath)
+        if err != nil {
+            return err
+        }
+
+        if length := len(stderror); stderror[length-3:length-1] == "OK" {
+            cmd.Println("Build Provenance verified successfully!")
+        } else {
+            cmd.Println("Unsuccessful verification of build provenance")
+        }
+    }
+
+    if !disableDeletes {
+        destDir = strings.TrimSuffix(destDir, "/metadata.json")
+        if err := os.RemoveAll(destDir); err != nil {
+            return err
+        }
+    }
+    return nil
+}
+
+func verifyNONPremiumPackage(cmd *cobra.Command, destDir, serviceAccountKeyFilePath, artifactPath, language, packageID, version string, disableDeletes, verifyBuildProvenance, disableCertificateVerification bool) error {
```

Comment:
inputs can be in a single struct. And for package as well, check if something from this function and the previous function can be abstracted out.

Reply:
done